### PR TITLE
"+" UIBarButton item  to add list

### PR DESCRIPTION
**Pull request by pravalgautam**
Implements a "+" from #4188

### CODE REVIEW

**Comment by pravalgautam** [comment]
Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 43 50

**Comment by biodranik** [comment]
Please create separate PRs for separate fixes/features.

**Comment by biodranik** [comment]
The bottom button "Create a new list" can also stay in addition to a + button. On large modern devices, it's more convenient to tap a button at the bottom.

The plan is also to add the "Export all bookmarks" button at the bottom, so both buttons can sit together nicely.

**Comment by pravalgautam** [comment]
> The bottom button "Create a new list" can also stay in addition to a + button. On large modern devices, it's more convenient to tap a button at the bottom.
> 
> The plan is also to add the "Export all bookmarks" button at the bottom, so both buttons can sit together nicely.

Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 16 05 27

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 1]
```diff
@@ -202,7 +202,7 @@ final class PlacePagePreviewViewController: UIViewController {
         details = nil;
       }
       
-      setScheduleLabel(state: L("closed_now"),
+      setScheduleLabel(state: L(""),
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:205]: Is it from another PR?

Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:205]: yes

Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:205]: this  is the PR belong to this isssue https://github.com/organicmaps/organicmaps/pull/4491/commits/cbfc515a8c821f5042f8341dc5eb2b10ef095db5

Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:205]: this PR is cbfc515a8c821f5042f8341dc5eb2b10ef095db5

**Review by vng — COMMENTED** [review]
@pravalgautam @biodranik 
Next steps here? LGTM

File: xcode/shaders/shaders.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -176,7 +176,7 @@
 		45789EE321353CA3009955CC /* program_manager_metal.mm */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.objcpp; path = program_manager_metal.mm; sourceTree = "<group>"; };
 		45789EE52135464D009955CC /* metal_program_params.hpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.h; path = metal_program_params.hpp; sourceTree = "<group>"; };
 		45789EE62135464D009955CC /* metal_program_params.mm */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.objcpp; path = metal_program_params.mm; sourceTree = "<group>"; };
-		4598437C21394BE000F8CAB2 /* shaders_metal.metallib */ = {isa = PBXFileReference; includeInIndex = 0; lastKnownFileType = "archive.metal-library"; path = shaders_metal.metallib; sourceTree = BUILT_PRODUCTS_DIR; };
+		4598437C21394BE000F8CAB2 /* shaders_metal.metallib */ = {isa = PBXFileReference; explicitFileType = "archive.metal-library"; includeInIndex = 0; path = shaders_metal.metallib; sourceTree = BUILT_PRODUCTS_DIR; };
```
Comment (biodranik) [inline xcode/shaders/shaders.xcodeproj/project.pbxproj:179]: Why is this change necessary?

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 3]
```diff
@@ -1,242 +0,0 @@
-final class PlacePageDirectionView: UIView {
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:1]: Why is this file deleted?

File: iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift [hunk 4]
```diff
@@ -117,8 +117,8 @@ extension BMCDefaultViewModel {
       return
     }
 
-    categories.append(manager.category(withId: manager.createCategory(withName: name)))
-    view?.insert(at: [IndexPath(row: categories.count - 1, section: section)])
+    categories.insert(manager.category(withId: manager.createCategory(withName: name)), at: 0)
```
Comment (biodranik) [inline iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift:120]: Why is this change necessary?

**Comment by pravalgautam** [comment]
> Can you please write "Closes #issuenumber" in the PR description, to reference the issue that you're fixing?

 I am sorry for the clutter I have made 
the commit that I fixed f874c2ed28495b3fd484bb1166173bfd2080d62c
I fixed the  and als I am sorry for the clutter I have made 
the commit that I fixed f874c2ed28495b3fd484bb1166173bfd2080d62c
mentioned issuenumber  the and signed (DCO) also

**Comment by pravalgautam** [comment]
> Can you please write "Closes #issuenumber" in the PR description, to reference the issue that you're fixing?

 I am sorry for the clutter that  I have made
solved issue #4188 commit --https://github.com/organicmaps/organicmaps/commit/f874c2ed28495b3fd484bb1166173bfd2080d62c
mentioned issuenumber the and signed (DCO) also

File: iphone/Maps/Bookmarks/Categories/BMCView/BMCViewController.swift [hunk 5]
```diff
@@ -35,6 +35,12 @@ final class BMCViewController: MWMViewController {
   override func viewDidLoad() {
     super.viewDidLoad()
     viewModel = BMCDefaultViewModel()
+    
+    let addButton = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addTapped))
+    navigationItem.rightBarButtonItem = addButton
+  }
+    @objc func addTapped() {
+    createNewCategory()
```
Comment (biodranik) [inline iphone/Maps/Bookmarks/Categories/BMCView/BMCViewController.swift:43]: Will this work? (you likely need to add `@objc` annotation to createNewCategory() method.

```suggestion
    navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(createNewCategory))
```

**Review by biodranik — COMMENTED** [review]
I've updated the PR description.

Did you test list deletion and OM restarting? Does the order of lists stay the same?

**Comment by pravalgautam** [comment]
> I've updated the PR description.
> 
> Did you test list deletion and OM restarting? Does the order of lists stay the same?

I need to fix this after relaunching all the lists are deleted

**Comment by pravalgautam** [comment]
> I've updated the PR description.
> 
> Did you test list deletion and OM restarting? Does the order of lists stay the same?

yes I test deletion and the lists stay the same  but Resolved -- if we add  lists and directly discard the app(using swipe up ) without using the navigation button then the lists disappeared too
the commit for this---11bab0224195d04170979f006146c0f8b448a5ba

**Review by biodranik — CHANGES_REQUESTED** [review]
The PR looks messed up. Did you learn about and try to git rebase and git commit --amend?

Let's split this issue into two. Please leave only adding of a + button in this PR. And create a new PR for the sorting issue.

**Comment by pravalgautam** [comment]
> The PR looks messed up. Did you learn about and try to git rebase and git commit --amend?
> 
> Let's split this issue into two. Please leave only adding of a + button in this PR. And create a new PR for the sorting issue.

Yes, I learned about the rebase but sir do I have to make an issue  and then pull a request a PR to it

File: iphone/Maps/Bookmarks/Categories/BMCView/BMCViewController.swift (hunk 5 above)
Comment (biodranik) [reply iphone/Maps/Bookmarks/Categories/BMCView/BMCViewController.swift:43]: Did you test it?

File: iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift (hunk 4 above)
Comment (biodranik) [inline iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift:120]: As I understood from your testing, there are some issues with this approach, right?

Let's remove these lines from this PR so it can be merged. And implement sorting in another PR.

**Review by biodranik — CHANGES_REQUESTED** [review]

**Comment by biodranik** [comment]
Do you see open unresolved comments?

**Comment by pravalgautam** [comment]
> Do you see open unresolved comments?

yes
